apps/passport-server/test/gi/genericIssuance.spec.ts

```diff
@@ -0,0 +1,687 @@
+import { getEdDSAPublicKey } from "@pcd/eddsa-pcd";
+import {
+  EdDSATicketPCD,
+  EdDSATicketPCDPackage,
+  expectIsEdDSATicketPCD
+} from "@pcd/eddsa-ticket-pcd";
+import { EmailPCDPackage } from "@pcd/email-pcd";
+import {
+  FeedCredentialPayload,
+  GenericIssuanceCheckInResult,
+  InfoResult,
+  PipelineFeedInfo,
+  PollFeedResult,
+  createFeedCredentialPayload,
+  createGenericCheckinCredentialPayload,
+  requestGenericIssuanceCheckin,
+  requestPipelineInfo,
+  requestPollFeed
+} from "@pcd/passport-interface";
+import { expectIsReplaceInFolderAction } from "@pcd/pcd-collection";
+import { ArgumentTypeName, SerializedPCD } from "@pcd/pcd-types";
+import { SemaphoreIdentityPCDPackage } from "@pcd/semaphore-identity-pcd";
+import {
+  SemaphoreSignaturePCD,
+  SemaphoreSignaturePCDPackage
+} from "@pcd/semaphore-signature-pcd";
+import { Identity } from "@semaphore-protocol/identity";
+import { expect } from "chai";
+import { randomUUID } from "crypto";
+import "mocha";
+import { SetupServer } from "msw/node";
+import * as path from "path";
+import {
+  ILemonadeAPI,
+  LemonadeTicket,
+  LemonadeTicketTier,
+  LemonadeUser
+} from "../../src/apis/lemonade/lemonadeAPI";
+import { getI18nString } from "../../src/apis/pretix/genericPretixAPI";
+import { stopApplication } from "../../src/application";
+import { PipelineDefinitionDB } from "../../src/database/queries/pipelineDefinitionDB";
+import { sqlQuery } from "../../src/database/sqlQuery";
+import { GenericIssuanceService } from "../../src/services/generic-issuance/genericIssuanceService";
+import {
+  LemonadePipeline,
+  LemonadePipelineDefinition
+} from "../../src/services/generic-issuance/pipelines/LemonadePipeline";
+import {
+  PretixPipeline,
+  PretixPipelineDefinition
+} from "../../src/services/generic-issuance/pipelines/PretixPipeline";
+import {
+  Pipeline,
+  PipelineDefinition,
+  PipelineType
+} from "../../src/services/generic-issuance/pipelines/types";
+import { Zupass } from "../../src/types";
+import { LemonadeDataMocker } from "../lemonade/LemonadeDataMocker";
+import { MockLemonadeAPI } from "../lemonade/MockLemonadeAPI";
+import { GenericPretixDataMocker } from "../pretix/GenericPretixDataMocker";
+import { getGenericMockPretixAPIServer } from "../pretix/MockGenericPretixServer";
+import { overrideEnvironment, testingEnv } from "../util/env";
+import { startTestingApp } from "../util/startTestingApplication";
+import {
+  expectFalse,
+  expectLength,
+  expectToExist,
+  expectTrue
+} from "../util/util";
+
+/**
+ * TODO: extract this to the `@pcd/passport-interface` package.
```

Comment:
This is substantially the same code as https://github.com/proofcarryingdata/zupass/blob/118302409a3f819d4eea9bc47fe1a9d837c2c89c/packages/lib/passport-interface/src/CredentialManager.ts#L177

It might be possible to just make that method public, or turn it into a function (or, it might make sense to instantiate a `CredentialManager` here and use `requestCredential` for the whole credential generation process).